diff --git a/packages/openbridge-webcomponents/src/components/modal-window/modal-window.ts b/packages/openbridge-webcomponents/src/components/modal-window/modal-window.ts
index 72c09a70..2d69ac14 100644
--- a/packages/openbridge-webcomponents/src/components/modal-window/modal-window.ts
+++ b/packages/openbridge-webcomponents/src/components/modal-window/modal-window.ts
@@ -38,6 +38,7 @@ export type ObcModalWindowDoneClickEvent = CustomEvent<void>;
 export class ObcModalWindow extends LitElement {
   @property({type: String}) size = ObcModalWindowSize.Large;
   @property({type: Boolean}) hasOptionalAction = false;
+  @property({type: Boolean}) hasCancelAction = true;
   @property({type: Boolean}) hasLeadingIcon = false;
 
   /**
@@ -110,12 +111,14 @@ export class ObcModalWindow extends LitElement {
               'primary-action-vertical': isSmall,
             })}
           >
-            <obc-button
-              @click=${this.onCancelClick}
-              .fullWidth=${isSmall || isMedium}
-            >
-              <slot name="cancel-label">Cancel</slot>
-            </obc-button>
+            ${this.hasCancelAction
+              ? html`<obc-button
+                  @click=${this.onCancelClick}
+                  .fullWidth=${isSmall || isMedium}
+                >
+                  <slot name="cancel-label">Cancel</slot>
+                </obc-button>`
+              : nothing}
             <obc-button
               variant="raised"
               @click=${this.onDoneClick}
